extension/src/experiments/index.ts

```diff
@@ -114,7 +114,7 @@ export class Experiments extends BaseRepository<TableData> {
     this.onDidChangeCheckpoints = this.checkpointsChanged.event
 
     this.experiments = this.dispose.track(
-      new ExperimentsModel(dvcRoot, workspaceState)
+      new ExperimentsModel(dvcRoot, workspaceState, internalCommands)
```

Comment:
[I] We shouldn't be calling `internalCommands` from inside of the model. Can we call `git log` with `git log --pretty=format:%B -n 3` or `git log HEAD~2..HEAD --pretty=format:%B` or equivalent and pass that into `transformAndSet`? That would do a few things: 

1. limit the number of times we call Git
2. stop us from having to mock the function/internal commands inside of the tests
3. not add to the responsibilities of the model

Reply:
Should be able to use `-z` to separate the messages with `null`s instead of new lines.

Reply:
Three calls every time we call `exp show` will clog up the output channel too: 
<img width="1346" alt="image" src="https://user-images.githubusercontent.com/37993418/210029466-19143f01-cefe-473c-9cd6-f4641185e37b.png">

Reply:
`git log --pretty="format:%H %B" -n 3 -z ` could be the way to go, that way you can collect an object that is keyed by the `sha`.

Reply:
> Should be able to use -z to separate the messages with nulls instead of new lines.

Thanks, I couldn't figure out a way to split the commit messages if they all came in one output and that flag works great!